python/TestHarness/TestHarness.py

```diff
@@ -444,240 +339,113 @@ def checkExpectError(self, output, expect_error):
         else:
             return True
 
-    # PBS Defs
-    def checkPBSEmulator(self):
-        try:
-            qstat_process = subprocess.Popen(['qstat', '--version'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-            qstat_output = qstat_process.communicate()
-        except OSError:
-            # qstat binary is not available
-            print('qstat not available. Perhaps you need to load the PBS module?')
-            sys.exit(1)
-        if len(qstat_output[1]):
-            # The PBS Emulator has no --version argument, and thus returns output to stderr
-            return True
+    # handleTestStatus is the entry point the testers will use to print statuses to the screen
+    def handleTestStatus(self, tester):
+        # The test has finished regardless of status
+        if tester.isFinished():
+            if tester.isSilent() or (tester.isDeleted() and not self.options.extra_info):
```

Comment:
Somewhere in the code, we're going to need to know when and when not to print. The scheduler does not decide this. It just throws a job in the status queue, which in turns calls back to the TestHarness.

Maybe we can always store the finalized output only in the tester... and _always_ print that object no matter what. That way the tester controls when/how/if stuff gets printed.
```python
def handleTestStatus(self, tester):
    footer = keep_track_of_totals(tester)
    if tester.getOutput() is not None:
        print tester.getOutput()

def printFinalResults(self):
    print footer
```